File: src/main/java/com/atlaspharmacy/atlaspharmacy/users/domain/valueobjects/AverageGrade.java
package com.atlaspharmacy.atlaspharmacy.users.domain.valueobjects;

import javax.persistence.Embeddable;

@Embeddable
public class AverageGrade {
    private double excellent;
    private double veryGood;
    private double good;
    private double poor;
    private double veryPoor;

    public AverageGrade(){}
    
    public AverageGrade(double excellent, double veryGood, double good, double poor, double veryPoor) {
        this.excellent = excellent;
        this.veryGood = veryGood;
        this.good = good;
        this.poor = poor;
        this.veryPoor = veryPoor;
    }


    public double count() {
        return ((5 * excellent) + (4 * veryGood) + (3 * good) + (2 * poor) + (1 * veryPoor))
                / (excellent + veryGood + good + poor + veryPoor);

    }

    public double getExcellent() { return excellent; }

    public void setExcellent(double excellent) { excellent = excellent; }

    public double getVeryGood() {
        return veryGood;
    }

    public void setVeryGood(double veryGood) {
        veryGood = veryGood;
    }

    public double getGood() {
        return good;
    }

    public void setGood(double good) {
        good = good;
    }

    public double getPoor() {
        return poor;
    }

    public void setPoor(double poor) {
        poor = poor;
    }

    public double getVeryPoor() {
        return veryPoor;
    }

    public void setVeryPoor(double veryPoor) {
        veryPoor = veryPoor;
    }
}

File: frontend/src/app/model/users/averageGrade.ts
export class AverageGrade{
  public excellent:Number;
  public veryGood:Number;
  public good:Number;
  public poor:Number;
  public veryPoor:Number;

  constructor(excellent:Number,veryGood:Number,good:Number,poor:Number,veryPoor:Number) {
    this.excellent = excellent;
    this.veryGood = veryGood;
    this.good = good;
    this.veryPoor = veryPoor;
    this.poor = poor;
  }
  count() {
    return ((5 * this.excellent.valueOf()) + (4 * this.veryGood.valueOf()) + (3 * this.good.valueOf())
      + (2 * this.poor.valueOf()) + (1 * this.veryPoor.valueOf()))
      / (this.excellent.valueOf() + this.veryGood.valueOf() + this.good.valueOf()
        + this.poor.valueOf() + this.veryPoor.valueOf());

  }
}
